### PR TITLE
add unit tests for FusedLAMB optimizer

### PR DESCRIPTION
**Pull request by kexinyu**
Add reference LAMB implementation, which is a PyTorch equivalent to this TensorFlow addon
https://github.com/tensorflow/addons/blob/master/tensorflow_addons/optimizers/lamb.py + global grad norm clipping,
to test `FusedLAMB(use_nvlamb=True)`, i.e. apply adaptive learning rate to all parameters

### CODE REVIEW

**Comment by FDecaYed** [comment]
@kexinyu could you also add a test for weight_decay != 0 case?

**Comment by kexinyu** [comment]
> @kexinyu could you also add a test for weight_decay != 0 case?

Non-zero weight decay test cases added.
